### ISSUE TITLE
iOS App with shared Compose UI module crashes on launch with `MissingResourceException`

### ISSUE THREAD

**Issue by aaalaniz** [JetBrains/compose-multiplatform#5011]
**Describe the bug**

Initially filed as #4991, I thought the crash was an x86_64 issue, but after installing an [unhandled exception handler](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.native/set-unhandled-exception-hook.html) and updating my project's CI pipeline to run tests on an arm64 iOS simulator, I see the following error when launching the app.

```
org.jetbrains.compose.resources.MissingResourceException: Missing resource with path: /Users/runner/Library/Developer/CoreSimulator/Devices/C4607EA8-660F-460C-BB4E-121AC02D737C/data/Containers/Bundle/Application/0C87CAA6-D7E0-4BAC-B8C0-B025731BE589/Lightsaber.app/compose-resources/composeResources/lightsaber.shared.generated.resources/values/strings.commonMain.cvr
    at 0   Lightsaber                          0x1015fa1c7        kfun:kotlin.Throwable#<init>(kotlin.String?){} + 119 
    at 1   Lightsaber                          0x1015f383b        kfun:kotlin.Exception#<init>(kotlin.String?){} + 115 
    at 2   Lightsaber                          0x101eb8a9f        kfun:org.jetbrains.compose.resources.MissingResourceException#<init>(kotlin.String){} + 175 
    at 3   Lightsaber                          0x101ec5a4b        kfun:org.jetbrains.compose.resources.object-2.readData#internal.11 + 479 
    at 4   Lightsaber                          0x101ec4eb7        kfun:org.jetbrains.compose.resources.object<…>
```

This seems very similar to the issue @chrisbanes reported in #4995, but I was not able to figure out how to resolve the crash. I am unable to reproduce it locally when building from Fleet or Xcode.

**Affected platforms**
- iOS

**Versions**
- Libraries:
  - Compose Multiplatform version: `1.6.11`
- Kotlin version: `2.0.0`
- OS version(s): `MacOS 14 with Xcode 15.3`
- OS architecture: `arm64`
- Device (model or simulator for iOS issues):  `arm64`

**To Reproduce**

I have been unable to reproduce this locally, but this crash happens every time when I build my [side project](https://github.com/aaalaniz/compose-lightsaber) in a Github Action and run some simple Maestro tests on an arm64 iOS 17.2 simulator. I have a draft PR where I have been trying to root cause and resolve this crash [here](https://github.com/aaalaniz/compose-lightsaber/pull/24/files).

Essentially the steps to reproduce are copied from my project's CI pipeline.

1. `./gradlew shared:assemble`
2. `./gradlew podInstall`
3. `xcodebuild -workspace iosApp/iosApp.xcworkspace -scheme iosApp -configuration Debug -xcconfig iosApp/Configuration/Config.xcconfig OBJROOT=iosApp/build/ios SYMROOT=iosApp/build/ios -sdk iphonesimulator`
4. Install the .app file: `xcrun simctl install UDID iosApp/build/ios/Debug-iphonesimulator/Lightsaber.app`
5. Launch the app or in the case of CI, launch the maestro tests: `maestro test .maestro`

**Expected behavior**
The maestro tests pass locally when I build and deploy the app to my simulator, but the actual behavior is the app crashes on launch when run in GitHub Actions.

**Comment by aaalaniz** [comment]
I investigated more and found that the following build phase, `PhaseScriptExecution [CP]\ Copy\ Pods\ Resources` is not executed when the CI pipeline runs the `xcodebuild` command provided. This probably explains why the resources are not available at runtime. I think this phase was run automatically with the same build instructions before updating to Kotlin 2.0.0 and Compose Multiplatform 1.6.11. iOS build phases are not my strength, but I will keep digging and see if I can provide a workaround.

**Comment by aaalaniz** [comment]
I resolved the issue after referencing [this comment](https://github.com/JetBrains/compose-multiplatform/issues/4720#issuecomment-2100898893) in #4720 and updating the steps to reproduce to include the following.

1. In Github Action job 1:  `./gradlew shared:assemble`
2. In Github Action job 2:

```bash
mkdir -p ./shared/build/compose/cocoapods/compose-resources
./gradlew generateDummyFramework
./gradlew podInstall
```
3. `xcodebuild -workspace iosApp/iosApp.xcworkspace -scheme iosApp -configuration Debug -xcconfig iosApp/Configuration/Config.xcconfig OBJROOT=iosApp/build/ios SYMROOT=iosApp/build/ios -sdk iphonesimulator`
4. Install and run tests

Adding the `mkdir` and `generateDummyFramework` ensures that the resources are copied over as expected. I was a little surprised that this workaround ended up being the solution. I thought I had set up GitHub Action directory caching for `./shared/build`, so would expect that any compose-resources needed would have been created when `./gradlew shared:assemble` was run in the previous job and then copied over to the iOS app build job. Perhaps, I missed something though 🤷🏼  

Thank you for posting the workaround! 🙏🏼
